### PR TITLE
Billing Tests for SubscriptionManager

### PR DESCRIPTION
**Pull request by Kokobe**
Tests that ensure that the conditions that change `isBillingAvailableFlow` and `isPremiumPurchaseAvailableFlow`  in `SubscriptionManager` work correctly.  

### Pointers to help with review
* https://www.figma.com/file/v2g41XtVbKj0QWmPsq4luw/Design-critique---Android-Premium-Functional-Design-Flows?node-id=101%3A1524&t=EbEd1oeR6tFg4zog-0

### How was this tested?
- [ ] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [x] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [ ] I checked that `@Previews` still work in Android Studio

### CODE REVIEW

File: app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        context = ApplicationProvider.getApplicationContext()
+        sharedPreferencesModel = SharedPreferencesModel(context)
+        settingsDataModel = mock {
+            on { getSettingsToggleValue(SettingsToggle.DEBUG_ENABLE_BILLING) } doReturn true
+        }
+
+        productDetailsList = mockk {
+            coEvery { isNullOrEmpty() } coAnswers { false }
+        }
+        existingPurchasesList = listOf(mockk())
```
Comment (dan-alcantara-neeva) [inline app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt:88]: This looks kind of weird... are you intentionally returning lists that have a single mocked out object inside of them?

File: app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+    private lateinit var billingClientController: BillingClientController
+    private lateinit var context: Context
+    private lateinit var settingsDataModel: SettingsDataModel
+    private val testProductDetailsFlow = MutableStateFlow(
+        ProductDetailsWrapper(isSet = false)
+    )
+    private lateinit var productDetailsList: List<ProductDetails>
+    private lateinit var offers: List<SubscriptionOfferDetails>
+    private val testPurchasesFlow = MutableStateFlow<List<Purchase>?>(null)
```
Comment (dan-alcantara-neeva) [inline app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt:67]: Separate out the real `vals` from the mocked out ones.  Makes it easier to understand what you're setting manually in the tests.

File: app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt (hunk 1 above)
Comment (Kokobe) [reply app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt:88]: The Billing API doesn't let me instantiate `Purchase` objects without the proper json string, but testing with the correct string is and checking its correct or not is beyond the scope of this test

Comment (Kokobe) [reply app/src/test/java/com/neeva/app/billing/SubscriptionManagerTest.kt]: Ah i can just make an explicit `@Mock` variable for a `Purchase`, working on that now

**Review by dan-alcantara-neeva — APPROVED** [review]

**Pull request merged**